examples/scroll.py

```diff
@@ -151,29 +150,62 @@ def main(image_file=None):
         clock.tick()
 
         going = True
+
+        # scroll events
+        scroll_down = 0
+        scroll_up = 0
+        scroll_left = 0
+        scroll_right = 0
+
         while going:
             # wait for events before doing anything.
             # events = [pg.event.wait()] + pg.event.get()
             events = pg.event.get()
 
             for e in events:
+                # if a key is pressed, update the scroll event associated with it (or exit the program if it's escape)
                 if e.type == pg.KEYDOWN:
                     if e.key == pg.K_ESCAPE:
                         going = False
                     elif e.key == pg.K_DOWN:
-                        scroll_view(screen, image, DIR_DOWN, view_rect)
+                        scroll_down = 1
+                    elif e.key == pg.K_UP:
+                        scroll_up = 1
+                    elif e.key == pg.K_LEFT:
+                        scroll_left = 1
+                    elif e.key == pg.K_RIGHT:
+                        scroll_right = 1
+
+                # if a key is lifted, the scroll event associated with that key will be set to 0.
+                if e.type == pg.KEYUP:
+                    if e.key == pg.K_DOWN:
+                        scroll_down = 0
                     elif e.key == pg.K_UP:
-                        scroll_view(screen, image, DIR_UP, view_rect)
+                        scroll_up = 0
                     elif e.key == pg.K_LEFT:
-                        scroll_view(screen, image, DIR_LEFT, view_rect)
+                        scroll_left = 0
                     elif e.key == pg.K_RIGHT:
-                        scroll_view(screen, image, DIR_RIGHT, view_rect)
+                        scroll_right = 0
+                
+                # quit if the event is quit.
                 elif e.type == pg.QUIT:
                     going = False
+
+                # handle mouse button presses on arrows.
                 elif e.type == pg.MOUSEBUTTONDOWN:
                     direction = regions.get_at(e.pos)[0]
                 elif e.type == pg.MOUSEBUTTONUP:
                     direction = None
+            
+            # given a scroll event, preform that action. This enables concurrent presses and multidirectional scrolling.
+            if(scroll_up):
```

Comment:
For these if statements, it would be better to get rid of the parathesis to stay consistent with the other if statements in the code.